cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcClient.java

```diff
@@ -139,7 +227,7 @@ public OperationSnapshot createCluster(String name, String imageVersion, Map<Str
                      .build())
         .build();
 
-      return client.createClusterAsync(conf.getProjectId(), conf.getRegion(), cluster).getInitialFuture().get();
+      return client.createClusterAsync(projectId, conf.getRegion(), cluster).getInitialFuture().get();
```

Comment:
should we try to auto-detect the region too?

Reply:
yeah, the have zone information is returned by this endpoint `http://metadata.google.internal/computeMetadata/v1/instance/zone`, but region is not surfaced.